File: pset6/pset6/readability/readability.py
from cs50 import get_string


def main():  # def main function
    text = get_string("Text: ")
    letters = count_letters(text)
    print(f"{letters} letter(s)")
    words = count_words(text)
    print(f"{words} word(s)")
    sentences = count_sentences(text)
    print(f"{sentences} sentence(s)")
    L = float(letters) / float(words) * 100
    S = float(sentences) / float(words) * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    if(index > 16):
        print("Grade 16+")
    elif(index < 1):
        print("Before Grade 1")
    elif(index < 16 and index > 1):
        print(f"Grade {round(index)}")


def count_letters(text):
    letters = 0
    i = 0
    for i in range(i, len(text)):   # boucle to 0 char to len string
        if text[i].isalpha() == True:   # check if is alpha per each char
            letters += 1
            i += 1
    return letters


def count_words(text):
    words = 1
    i = 0
    for i in range(i, len(text)):   # boucle again to check
        if(" " == text[i]):   # check if is a space for count a word
            words += 1
            i += 1
    return words


def count_sentences(text):
    sentences = 0
    i = 0
    for i in range(i, len(text)):   # boucle again to check
        if("." == text[i] or "?" == text[i] or "!" == text[i]):  # check if its end of sentence
            sentences += 1
            i += 1
    return sentences


main()

File: pset2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>


int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("text: ");
    int letters = count_letters(text);
    printf("%i letter(s) \n", letters);
    int words = count_words(text);
    printf("%i word(s) \n", words);
    int sentences = count_sentences(text);
    printf("%i sentence(s) \n", sentences);
    float L = (float) letters / (float) words * 100;
    float S = (float) sentences / (float) words * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    if (index > 16)
    {
        printf("Grade 16+");
    }
    else if (index < 1)
    {
        printf("Before Grade 1");
    }
    else if (index < 16 && index > 1)
    {
        printf("Grade %i", (int) round(index));
    }
    printf("\n");
}

int count_letters(string text)
{
    // TODO: Compute and return score for string
    int letters = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isupper(text[i]))
        {
            letters += 1;
        }
        else if (islower(text[i]))
        {
            letters += 1;
        }
    }
    return letters;
}

int count_words(string text)
{
    // TODO: Compute and return score for string
    int words = 1;

    for (int j = 0, len = strlen(text); j < len; j++)
    {
        if (' ' == (text[j]))
        {
            words += 1;
        }
    }
    return words;
}

int count_sentences(string text)
{
    // TODO: Compute and return score for string
    int sentences = 0;

    for (int k = 0, len = strlen(text); k < len; k++)
    {
        if ('.' == (text[k]) || '?' == (text[k]) || '!' == (text[k]))
        {
            sentences += 1;
        }
    }
    return sentences;
}